# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/pine/hotfix.patch
# ROCK Linux is Copyright (C) 1998 - 2004 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./pine/osdep/debuging.tim.orig	2002-11-28 01:19:37.000000000 +0100
+++ ./pine/osdep/debuging.tim	2003-08-16 13:53:23.000000000 +0200
@@ -195,20 +195,22 @@
 {
     va_list args;
 
-    va_start(args, fmt);
-
-    if(debug_fp && debug >= ps_global->dlevel && do_debug(debug_fp))
-      vfprintf(debug_fp, fmt, args);
+    if(debug_fp && debug >= ps_global->dlevel && do_debug(debug_fp)) {
+        va_start(args, fmt);
+        vfprintf(debug_fp, fmt, args);
+        va_end(args);
+    }
 
     if(ps_global->dlevel <= 9 || ps_global->dlevel <= debug){
 	char    b[64000];
 
+        va_start(args, fmt);
 	vsnprintf(b, sizeof(b), fmt, args);
+        va_end(args);
+
 	b[sizeof(b)-1] = '\0';
 	add_review_message(b, ps_global->dlevel);
     }
-
-    va_end(args);
 }
 #endif /* DEBUGJOURNAL */
 
